khmer/_cpy_counttable.hh

```diff
@@ -0,0 +1,132 @@
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+LICENSE (END)
+
+Contact: khmer-project@idyll.org
+*/
+
+typedef struct {
+    khmer_KHashtable_Object khashtable;
+    Counttable * counttable;
+} khmer_KCounttable_Object;
+
+static PyMethodDef khmer_counttable_methods[] = {
+    {NULL, NULL, 0, NULL}           /* sentinel */
+};
+
+static PyObject* khmer_counttable_new(PyTypeObject * type, PyObject * args,
+                                      PyObject * kwds);
+
+static PyTypeObject khmer_KCounttable_Type
+CPYCHECKER_TYPE_OBJECT_FOR_TYPEDEF("khmer_KCounttable_Object")
+= {
+    PyVarObject_HEAD_INIT(NULL, 0) /* init & ob_size */
+    "_khmer.Counttable",             /* tp_name */
+    sizeof(khmer_KCounttable_Object), /* tp_basicsize */
+    0,                             /* tp_itemsize */
+    0,                             /*tp_dealloc*/
+    0,              /*tp_print*/
+    0,              /*tp_getattr*/
+    0,              /*tp_setattr*/
+    0,              /*tp_compare*/
+    0,              /*tp_repr*/
+    0,              /*tp_as_number*/
+    0,              /*tp_as_sequence*/
+    0,              /*tp_as_mapping*/
+    0,              /*tp_hash */
+    0,              /*tp_call*/
+    0,              /*tp_str*/
+    0,              /*tp_getattro*/
+    0,              /*tp_setattro*/
+    0,              /*tp_as_buffer*/
+    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,       /*tp_flags*/
+    "counttable object",           /* tp_doc */
+    0,                       /* tp_traverse */
+    0,                       /* tp_clear */
+    0,                       /* tp_richcompare */
+    0,                       /* tp_weaklistoffset */
+    0,                       /* tp_iter */
+    0,                       /* tp_iternext */
+    khmer_counttable_methods,   /* tp_methods */
+    0,                       /* tp_members */
+    0,                       /* tp_getset */
+    0,                       /* tp_base */
+    0,                       /* tp_dict */
+    0,                       /* tp_descr_get */
+    0,                       /* tp_descr_set */
+    0,                       /* tp_dictoffset */
+    0,                       /* tp_init */
+    0,                       /* tp_alloc */
+    khmer_counttable_new,    /* tp_new */
+};
+
+
+//
+// khmer_counttable_new
+//
+
+static PyObject* khmer_counttable_new(PyTypeObject * type, PyObject * args,
+                                      PyObject * kwds)
+{
+    khmer_KCounttable_Object * self;
+
+    self = (khmer_KCounttable_Object *)type->tp_alloc(type, 0);
+
+    if (self != NULL) {
+        WordLength k = 0;
+        PyListObject * sizes_list_o = NULL;
```

Comment:
PyArg_ParseTuple doesn't increase the reference count, and presumably the ref count can't decrease to zero while we are using an argument to this function.